tests/discover/parametrize.sh

```diff
@@ -27,15 +30,33 @@ rlJournalStart
         rlRun "REPO=fmf tmt run -r $plan_env $steps | tee output"
         rlAssertGrep 'url: https://github.com/teemtee/tmt' 'output'
         # No substitution should happen
-        rlRun "REPO=tmt tmt run -r $plan $steps | tee output" 2
+        rlRun "REPO=tmt tmt run -r $plan_noenv $steps | tee output" 2
         rlAssertGrep 'url: https://github.com/teemtee/${REPO}' 'output'
     rlPhaseEnd
 
     rlPhaseStartTest 'Undefined variable'
-        rlRun "tmt run -r $plan $steps | tee output" 2
+        rlRun "tmt run -r $plan_noenv $steps | tee output" 2
         rlAssertGrep 'url: https://github.com/teemtee/${REPO}' 'output'
     rlPhaseEnd
 
+    rlPhaseStartTest 'From context attribute'
+        rlRun "tmt run -r $plan_ctx $steps | tee output"
+        rlAssertGrep 'url: https://github.com/teemtee/tmt' 'output'
+    rlPhaseEnd
+
+    rlPhaseStartTest 'From --context command line option'
+        rlRun "tmt -c repo=tmt run -r $plan_noctx $steps | tee output"
+        rlAssertGrep 'url: https://github.com/teemtee/tmt' 'output'
+        # Precedence of option over context attribute
+        rlRun "tmt -c repo=fmf run -r $plan_ctx $steps | tee output"
+        rlAssertGrep 'url: https://github.com/teemtee/fmf' 'output'
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Undefined context'
+        rlRun "tmt run -r $plan_noctx $steps | tee output" 2
+        rlAssertGrep 'url: https://github.com/teemtee/${repo}' 'output'
```

Comment:
So for the undefined context the original content was:

    url: https://github.com/teemtee/$@{repo}

After the expansion it is

    url: https://github.com/teemtee/${repo}

I guess it could be dangerous to modify content if the context dimension name is not matched.

Reply:
I believe I have fixed that, although the required code is a bit more complicated.

Reply:
Just to add, previously we could erroneously expand environment variables using context. This should be also fixed now.

Reply:
I have added new test scenario covering this.